### DISCUSSION TITLE
`Widget` force a parentElement on its domElement

### DISCUSSION THREAD

**Discussion by valentinMachado** [iTowns/itowns#2098] [Q&A]
It's not possible to create a `Widget` without a parentElement since parentElement is initialized [here](https://github.com/iTowns/itowns/blob/a8dfb36cb6c86d0e5a9a9db41f8fb9feb9a7e9ab/src/Utils/gui/Widget.js#L10) and then manipulated [there](https://github.com/iTowns/itowns/blob/a8dfb36cb6c86d0e5a9a9db41f8fb9feb9a7e9ab/src/Utils/gui/Widget.js#L30). In my case I want to create a widget then use a generic function that will integrate it in the DOM which is not possible without "hacking" (ie give a parentElement then the line after do `widget.domElement.remove()`.

We could keep the actual behavior but just do the difference between `undefined` and `null` meaning when its `undefined` `view.domElement` is the parent but when it's `null` no parent is setted.

WDYT ?

**Comment by jailln** [comment]
In the same idea than #2005 my opinion is that the `Widget` constructor should not know its parent and it should be resposible to add itself to the parent. I think it should be done in an util method as you want to do it.
